src/main/java/seedu/eduke8/Parser.java

```diff
@@ -1,13 +1,27 @@
 package seedu.eduke8;
 
 
-import seedu.eduke8.quiz.QuizInterface;
+import seedu.eduke8.option.OptionInterface;
+import seedu.eduke8.question.QuestionInterface;
+import seedu.eduke8.quiz.Quiz;
+import seedu.eduke8.topic.TopicListInterface;
 import seedu.eduke8.ui.UiInterface;
 
+import java.util.ArrayList;
+
 public class Parser implements ParserInterface {
 
-    public void parseCommand(String userInput) {
+    public Parser() {
+    }
+
+    public void parseCommand(TopicListInterface topicList, String userInput) {
+        int numOfQuestions = 0;
+        String topicName = "";
         String[] commandArr = userInput.trim().split(" ", 0);
+        if (commandArr.length == 3 && commandArr[0].equals("quiz")) {
```

Comment:
Maybe we can move this into the "quiz" case next time so we don't check for the command twice

Reply:
Ah yes, pardon my folly, will adjust it in future. Thank you!